txtorcon/endpoints.py

```diff
@@ -298,10 +298,13 @@ def __init__(self, reactor, config, public_port,
         self.progress_listeners = []
 
         if self.hidden_service_dir is None:
-            self.hidden_service_dir = tempfile.mkdtemp(prefix='tortmp')
-            log.msg('Will delete "%s" at shutdown.' % self.hidden_service_dir)
-            delete = functools.partial(shutil.rmtree, self.hidden_service_dir)
-            reactor.addSystemEventTrigger('before', 'shutdown', delete)
+            self.add_hidden_service_dir()
+
+    def add_hidden_service_dir(self):
+        self.hidden_service_dir = tempfile.mkdtemp(prefix='tortmp')
```

Comment:
I'd rather not have a public method to do this, as it's an internal thing.
If it's not possible to do this without a helper method, at least start it with an underscore?